cmd/otel-allocator/Dockerfile

```diff
@@ -3,12 +3,12 @@ FROM golang:1.19 as builder
 
 WORKDIR /app
 
-# Copy go mod and sum files
-COPY go.mod go.sum ./
+# copy operator and allocator code
+COPY . .
```

Comment:
Can this change be restructured/avoided?  Doing this can cause code changes that do not change dependencies to invalidate the cached FS layer with dependencies already downloaded, which can make developing/testing with this `Dockerfile` annoying.

Reply:
`go mod download` will fail we not copy the code first because of the [new replacement directive]( https://github.com/open-telemetry/opentelemetry-operator/pull/1220/files#diff-35d0de98f88a0bce74a9813fa595b33ed20736a2167c94fad07476411c32422aR175)

Alternatively we have to drop dependency to the operator here again.

Reply:
IMO we should drop the dependency on the operator, I think it's going to add more complexity than it's worth.